src/ngraph/runtime/cpu/cpu_emitter.cpp

```diff
@@ -4272,186 +4272,217 @@ namespace ngraph
             template <>
             void CPU_Emitter::EMITTER_DECL(ngraph::op::Softmax)
             {
-                writer.block_begin();
-                const ngraph::op::Softmax* softmax = static_cast<const ngraph::op::Softmax*>(node);
-                auto type = out[0].get_type();
-                auto shape = out[0].get_shape();
-                auto dims = out[0].get_shape().size();
-                auto axes = softmax->get_axes();
-
-                // create arg/out if 1d
-                if (dims < 1)
+                if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node))
```

Comment:
do we have tests for both 2d and 4d softmax?

Reply:
currently MKLDNN support softmax across single axis. no support for softmax across multiple axis yet. so dint yet add those test case